pkg/crc/machine/vsockprivateaddress_windows.go

```diff
@@ -0,0 +1,65 @@
+package machine
+
+import (
+	"errors"
+	"net"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+)
+
+const (
+	fallbackPrivateAddress = "127.0.0.1"
+	adapterName            = "vEthernet (WSL)"
+	ipv4Prefix             = "172."
+)
+
+// On Windows it would be useful if WSL2 can access VSock resources as well
+// This address should work on both the windows side and the WSL2 side
+
+var (
+	vsockCheckedWsl          = false
+	vsockPrivateAddressToUse = "127.0.0.1"
+)
+
+func VsockPrivateAddress() (addrStr string) {
+	if vsockCheckedWsl {
+		return vsockPrivateAddressToUse
+	}
+
+	addrResult, err := localhostOrWslhost()
+	if err != nil {
+		logging.Warnf("Encountered error trying to resolve WSL host IP: %s", err)
+	}
+	vsockCheckedWsl, vsockPrivateAddressToUse = true, addrResult
+	return addrResult
+}
+
+func localhostOrWslhost() (addrStr string, err error) {
+	interfaces, errIf := net.Interfaces()
+	if errIf != nil {
+		return fallbackPrivateAddress, errIf
+	}
+	for _, iface := range interfaces {
+		if iface.Name == adapterName {
+			addresses, errAddrs := iface.Addrs()
+			if errAddrs != nil {
+				return fallbackPrivateAddress, errAddrs
+			}
+			for _, address := range addresses {
+				addr := address.String()
+				if strings.HasPrefix(addr, ipv4Prefix) {
+					indexOfSlash := strings.IndexAny(addr, "/")
+					if indexOfSlash > 0 {
+						var addrIPOnly = addr[0:indexOfSlash] // We can simply strip the / as WSL2 sits at the start of the network
+						logging.Infof("private vsock binds will be to the WSL2 host address %s", addrIPOnly)
```

Comment:
I think this log can be a `debug` log, during start we already notify which IP address is assigned to the vm